src/Network/Transport/TCP.hs

```diff
@@ -1717,7 +1743,9 @@ runScheduledAction (ourEndPoint, theirEndPoint) mvar = do
     handleIOException ex vst = do
       -- Release probing resources if probing.
       forM_ (remoteProbing vst) id
-      tryCloseSocket (remoteSocket vst)
```

Comment:
I swapped this for `tryShutdownSocketBoth` which should cause `handleIncomingMessages` to throw an exception when trying to receive.